### PR TITLE
Set up sbt-ci-release to publish from GitHub Actions

### PR DESCRIPTION
**Pull request by olafurpg**
Problem: currently, the twitter/util release process requires several
manual steps.

Solution: This PR updates the sbt build to use the sbt-ci-release plugin
(https://github.com/olafurpg/sbt-ci-release/) to publish releases from
GitHub Actions.

In order for the CI releases to succeed, you will need to update the
secrets in the GitHub settings here
https://github.com/twitter/util/settings/secrets following
instructions from the readme here
https://github.com/olafurpg/sbt-ci-release/

Result: Every merge into the develop and master branches are
published as SNAPSHOT releases and every git tag push is published as a
stable release to Maven Central.

Caveat: The biggest change in the build is that the version number is now
automatically derived from the git tag using sbt-dynver
(https://github.com/dwijnand/sbt-dynver). The benefit of deriving the
version number from git is that it's possible to trigger a stable
release like 20.4.2 by pushing a tag named "v20.4.2". The current
naming convention for git tags is "util-$VERSION", which is not supported
by sbt-dynver. Stable releases from now on need to use the format "v$VERSION"
instead.

When working locally, this change means that the version number of the
sbt build dynamically changes based on the current git commit. It's
possible to manually set a stable version for an active sbt session with
the command `set every version := "20.4.4"`, in case you need that
version for manual `publishLocal` steps.

cc/ @cacoco

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+on:
+  push:
+    branches: [develop, master]
+    tags: ["v*"]
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: olafurpg/setup-scala@v7
```
Comment (olafurpg) [inline .github/workflows/release.yml:11]: The default Java version is the latest OpenJDK 8 HotSpot version via AdoptOpenJDK, it's possible to select another JDK as documented here https://github.com/olafurpg/setup-scala#usage For example to use JDK 11
```diff
      - uses: olafurpg/setup-scala@v7
+       with:
+         java-version: adopt@1.11
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+name: Release
+on:
+  push:
+    branches: [develop, master]
```
[the remark this answers is not in the document]
Comment (cacoco) [reply .github/workflows/release.yml:4]: @olafurpg we push to develop hourly, so it is likely we don't want to trigger publishing a snapshot at that frequency -- I think we want to have a process do it nightly. 

However, we can also discuss going back to pushing develop on a daily basis instead if the result is a snapshot getting published. I think we'd trade the frequency for that behavior.

[the remark this answers is not in the document]
Comment (olafurpg) [reply .github/workflows/release.yml:4]: I think GitHub Actions support running cron jobs, which should make it possible to publish a nightly SNAPSHOT.

**Comment by olafurpg** [comment]
Closing since this PR was only for demonstration purposes.

**Pull request closed without merging**